pkg/parser/ast.go

```diff
@@ -8,44 +8,72 @@ import (
 	"foxygo.at/evy/pkg/lexer"
 )
 
+// Node represents a node in the AST.
 type Node interface {
+	// Token returns the token of the Evy source program associated with the node.
```

Comment:
yes, maybe, maybe not. The evaluator also uses Token (and also for error reporting) - for panic or runtime errors. The entire token is then highlighted in the frontend.
There it could arguably also be a location. Token somehow feels more correct to me. It also seems that it would be more tractable if/when things go wrong (I had location errors before with the error highlights).

I'm not sure that this needs changing at this stage.

I don't think this needs tracking right now imo.

As I reworked the docs I have updated everything that I felt needed updating.

For now I'd like to keep issues empty or for "real" issues as the pop up.

If you want we could start a :List of camh's concerns" in the wiki of this repo?

Reply:
ok, no worries